diff -Naur fortune-mod-1.99.1/fortune/Makefile fortune-mod-1.99.1.oden/fortune/Makefile
--- fortune-mod-1.99.1/fortune/Makefile	2004-03-05 14:29:56.000000000 +0100
+++ fortune-mod-1.99.1.oden/fortune/Makefile	2008-12-25 09:45:41.000000000 +0100
@@ -8,7 +8,7 @@
 all: fortune
 
 fortune: fortune.o
-	$(CC) $(LDFLAGS) -o fortune fortune.o $(LIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o fortune fortune.o $(LIBS)
 
 clean:
 	rm -f fortune.o fortune fortune.man
diff -Naur fortune-mod-1.99.1/Makefile fortune-mod-1.99.1.oden/Makefile
--- fortune-mod-1.99.1/Makefile	2004-03-05 14:29:56.000000000 +0100
+++ fortune-mod-1.99.1.oden/Makefile	2008-12-25 09:45:17.000000000 +0100
@@ -65,7 +65,7 @@
 RECODELIBS=-lrecode
 
 DEFINES=-DFORTDIR="\"$(COOKIEDIR)\"" -DOFFDIR="\"$(OCOOKIEDIR)\"" -DLOCFORTDIR="\"$(LOCALDIR)\"" -DLOCOFFDIR="\"$(LOCALODIR)\""
-CFLAGS=-O2 $(DEFINES) -Wall -fomit-frame-pointer -pipe -fsigned-char
+CFLAGS=$(RPM_OPT_FLAGS) $(DEFINES)
 LDFLAGS=-s
 
 # The above flags are used by default; the debug flags are used when make
diff -Naur fortune-mod-1.99.1/util/Makefile fortune-mod-1.99.1.oden/util/Makefile
--- fortune-mod-1.99.1/util/Makefile	2004-03-05 14:29:57.000000000 +0100
+++ fortune-mod-1.99.1.oden/util/Makefile	2008-12-25 09:46:17.000000000 +0100
@@ -8,16 +8,16 @@
 all: strfile unstr rot
 
 strfile: strfile.o
-	$(CC) $(LDFLAGS) -o strfile strfile.o
+	$(CC) $(CFLAGS) $(LDFLAGS) -o strfile strfile.o
 
 unstr: unstr.o
-	$(CC) $(LDFLAGS) -o unstr unstr.o
+	$(CC) $(CFLAGS) $(LDFLAGS) -o unstr unstr.o
 
 randstr: randstr.o
-	$(CC) $(LDFLAGS) -o randstr randstr.o
+	$(CC) $(CFLAGS) $(LDFLAGS) -o randstr randstr.o
 
 rot: rot.o
-	$(CC) -fsigned-char $(LDFLAGS) -o rot rot.o
+	$(CC) $(CFLAGS) -fsigned-char $(LDFLAGS) -o rot rot.o
 
 clean:
 	rm -f *.o unstr strfile randstr rot ansify
